### ISSUE TITLE
Infinity loop crash when sending a tracking event property with a single-value array, e.g., ["test": [[1]]]

### ISSUE THREAD

**Issue by datoutwo** [amplitude/Amplitude-Swift#239]
## Expected Behavior
- Should not crash when event property's value is one single-value in the array.  

## Current Behavior
- App crash in the infinity loop of extension. 
```swift 
extension UnkeyedDecodingContainer {
    mutating func decode(_ type: [Any].Type) throws -> [Any] {
```

## Possible Solution
- Flat data if it just one value

## Steps to Reproduce
 

1. Change track event to `Amplitude.testInstance.track(eventType: eventType, eventProperties: ["test": [[1]]])` 
2. Send the event.  
<img width="700" alt="image" src="https://github.com/user-attachments/assets/c35cf20b-9b41-4151-b6d0-2cd511a55b9c">

## Reproducible example project 
 

1. Open `AmplitudeSwiftUIExample.xcodeproj` in the attach file. 
2. Run in xcode. 
3. Click button in the `GENERAL EVENT`. 
4. Few seconds app crash in the UnkeyedDecodingContainer decode function. 

[Amplitude-Swift.zip](https://github.com/user-attachments/files/17702622/Amplitude-Swift.zip)

## Environment
- SDK Version: v1.10.1
- OS Info:  Xcode 15.4 
- iOS Version: iPhone 15 pro 17.4 (every version should crash)

**Comment by crleona** [comment]
Hi @datoutwo, the example you've provided does not reproduce for me after checking with multiple iOS versions and multiple versions of Xcode.  Are you able to provide a more specific reproduction case?

**Comment by crleona** [comment]
Hi @datoutwo, thank you for the report and the PR. I've added another fix that also adds some tests in https://github.com/amplitude/Amplitude-Swift/pull/242. Could you please confirm that this also fixes the issue on your end?

**Comment by datoutwo** [comment]
Hi @crleona just check the pr it works like a charm from my side. Thanks for the speedy reply and fix.

**Comment by crleona** [comment]
The fix is released in [1.10.2](https://github.com/amplitude/Amplitude-Swift/releases/tag/v1.10.2).
